images/assets/patches/0013-Add-uvloop-as-the-the-Gunicorn-worker-eventloop.patch

```diff
@@ -0,0 +1,46 @@
+From 427fa5ffb225d03f5433bab3916d4c09c1057b14 Mon Sep 17 00:00:00 2001
+From: Pedro Brochado <pedropsb95@gmail.com>
+Date: Thu, 14 Nov 2024 03:53:56 -0300
+Subject: [PATCH] Add uvloop as the the Gunicorn worker eventloop
+
+---
+ pulpcore/content/entrypoint.py | 6 +++++-
+ pyproject.toml                 | 1 +
+ 2 files changed, 6 insertions(+), 1 deletion(-)
+
+diff --git a/pulpcore/content/entrypoint.py b/pulpcore/content/entrypoint.py
+index 5ae79cfe2..8d7effd08 100644
+--- a/pulpcore/content/entrypoint.py
++++ b/pulpcore/content/entrypoint.py
+@@ -1,11 +1,15 @@
+ import click
+ from pulpcore.app.pulpcore_gunicorn_application import PulpcoreGunicornApplication
++import os
+ 
+ 
+ class PulpcoreContentApplication(PulpcoreGunicornApplication):
+     def load_app_specific_config(self):
++        worker_class = "aiohttp.GunicornWebWorker"
++        if os.getenv("PULP_USE_UVLOOP"):
++            worker_class = "aiohttp.GunicornUVLoopWebWorker"
+         self.set_option("default_proc_name", "pulpcore-content", enforced=True)
+-        self.set_option("worker_class", "aiohttp.GunicornWebWorker", enforced=True)
++        self.set_option("worker_class", worker_class, enforced=True)
+ 
+     def load(self):
+         import pulpcore.content
+diff --git a/pyproject.toml b/pyproject.toml
+index 3e49e93ab..97262ef84 100644
+--- a/pyproject.toml
++++ b/pyproject.toml
+@@ -66,6 +66,7 @@ dependencies = [
+   "tablib<3.6.0",
+   "url-normalize>=1.4.3,<=1.4.3",
+   "uuid6>=2023.5.2,<=2024.7.10",
++  "uvloop>=0.21.0,<0.22",
+   "whitenoise>=5.0,<6.9.0",
+   "yarl>=1.8,<1.18.1",
+ ]
+-- 
+2.47.0
+
```

Comment:
Would you be ok if this part of the diff is pulled out since we're declaring the dependency directly below?

Reply:
Yes. In fact I was supposed to ask where was the right place at the time but I forget about it. Thanks for the input